### ISSUE TITLE
Rodrick the Reader card does not appear to work

### ISSUE THREAD

**Issue by mrhappyasthma** [Longwelwind/swords-and-ravens#1427]
I played this card and won a battle. I selected 'Clash of Kings' on Deck 2. However when the round ended, the next turn did not choose that card for the Westeros 2 deck.

Game: https://swordsandravens.net/play/c3c12510-6ac9-4ec6-b34f-af6cdf34daa8

Where I won:
https://i.imgur.com/IcHnGJc.png

What we actually got next round: 
https://i.imgur.com/2ZfA2kR.png

**Comment by gereon77** [comment]
Hi Mark! Thanks for your feedback. I will retest Rodrik again but I really think you by accident selected game of thrones. We have the dwd cards for a year now and Rodrik has been played a lot already. I guess we would have received more reports if he wouldn’t work. I also don’t know any side effect which could cause this.

Here you can check the code yourself: looks pretty straight forward:

https://github.com/Longwelwind/swords-and-ravens/blob/master/agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/post-combat-game-state/after-winner-determination-game-state/rodrik-the-reader-ability-game-state/RodrikTheReaderAbilityGameState.ts

Line 70ff

**Comment by mrhappyasthma** [comment]
I suppose it's possible i selected the wrong card, but I'm fairly certain I did not. I spent a lot of time deciding what action to take on that turn.

Is there anywhere in the game logs that track what was selected to confirm this?

**Comment by mrhappyasthma** [comment]
I see in the logs this: `Rodrik the Reader: House Greyjoy decided to choose a card from Westeros Deck 2.` but I'm not sure if we store and more logs server side.

Also idk if we support 'private' messages in the gamelog, but it might be cool to show that in the log for the person who played the card (but not for all other players). If this is functionality that is supported.

**Comment by gereon77** [comment]
> Is there anywhere in the game logs that track what was selected to confirm this?

Unfortunately no :/

**Comment by mrhappyasthma** [comment]
Well it's possible I somehow made a mistake or fat-fingered the button click or something. But I would have bet I selected the right card.

So maybe the only action items here would be:

1. Double check the logic in a test game. But it seems like no1 else has reported this issue so perhaps it's not a bug and it was human error.

2. (Ideally) Have some kind of private log of this. I imagine something like when you use the Raven to peek on the Wildlings deck. The player can mouse over the deck to re-check the card. It would be cool for something similar here.

**Comment by gereon77** [comment]
So, I just checked again our internal logs. As we do

`console.log(`Message for game ${entireGame.id} from user ${user.id}:\n\t${JSON.stringify(message)}`);` in `GlobalServer.ts` there are basic possibilities to retrieve some logs from the server. Unfortunately you used Rodrik on 28th of Feb and we have restarted the server several times since then. Therefore unfortunately we have no chance to check what you've sent. Indeed, logging this stuff permanently would be useful now, but those kind of logs would increase rapidly...

Later I will test Rodrik again and check the circumstances of your game. But on the first view I see nothing which could have changed the Westeros decks again.
